admin_ui/app/org/OrgEditPermissions.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from '@reach/router';
+import Store from '../Store';
+import OrgEditPermissionsModals from './OrgEditPermissionsModals';
+import useDocumentTitle from '@rehooks/document-title';
+import TabLinks from '../../components/TabLinks';
+import Button from '../../components/Button';
+import Grid from '../../components/Grid';
+
+// Dummy data
+let permissionHeadings = [
+  {
+    label: 'Name',
+    className: 'text-left',
+  },
+  { label: 'Is system?' },
+  { label: 'Role assignments' },
+  { label: 'Actions', isSortable: false, className: 'text-right' },
+];
+let permissionData = [
+  {
+    id: 7,
+    name: 'blog.read',
+    systemPermission: false,
+    roles: 1,
+  },
+  {
+    id: 8,
+    name: 'blog.modify',
+    systemPermission: false,
+    roles: 1,
+  },
+];
+
+const OrgEditPermissions = ({ orgId }) => {
+  // Load the store (we need access to store.org.currentModal$)
+  const store = React.useContext(Store);
+
+  // Set page title
+  useDocumentTitle(`Organization name: Permissions`);
+
+  return (
+    <React.Fragment>
+      <OrgEditPermissionsModals />
+      <h1 className="mb-6">&quot;Organization name&quot;: Permissions</h1>
+      <TabLinks
+        className="mb-6"
+        links={[
+          {
+            label: 'Org details',
+            to: `/organizations/${orgId}/edit`,
+          },
+          {
+            label: 'Permissions',
+            to: `/organizations/${orgId}/edit/permissions`,
+          },
+          {
+            label: 'Roles',
+            to: `/organizations/${orgId}/edit/roles`,
+          },
+          {
+            label: 'Users',
+            to: `/organizations/${orgId}/edit/users`,
+          },
+        ]}
+      />
+      <fieldset className="mb-6">
+        <legend>New permissions</legend>
+        <Button
+          onClick={() => {
+            store.org.currentPermissionsModal$.next('CREATE');
+          }}
+        >
+          Create new permission
+        </Button>
+        <p className="mt-4">
+          Tip: If you want to create a permission that is <em>not</em> scoped to the
+          &quot;ORGNAME&quot; organization, please{' '}
+          <Link to="/permissions">visit the permissions page</Link>.
+        </p>
+      </fieldset>
+      <fieldset className="mb-6">
+        <legend>Existing permissions</legend>
+        <Grid
+          headings={permissionHeadings}
+          data={permissionData}
+          renderer={(permissionData, index) => {
+            return (
+              <tr key={`permissionRow${index}`} className={index % 2 ? `bg-grey-lightest` : ``}>
+                <td className="p-2">
+                  <Link
+                    to={`/permissions/${permissionData.id}/edit`}
+                    onClick={(e) => {
```

Comment:
Hmmm, would creating a memoized callback really help us here?

Each link that would call such a `handleClick` method would need to pass `id` as well as `type` (with values `EDIT` or `DELETE`). We would also need to pass the current modal in the array of dependencies no?

Example:
```javascript
  // Memoize the handleClick handler (Called from "Edit" or "Delete" actions below)
  const handleClick = React.useCallback((e, id clickType) => {
// ... Do stuff in here with store.org.currentPermissionsModal$ depending on `clickType` param (as well as id?)
  }, [store.org.currentPermissionsModal$.getValue()]);```

Reply:
We'll resolve this for now and revisit once we have the actual modals working. Suggestion from @jmike : Instead of passing id, clickType have them as attributes of the HTML element and load them using `e.target.XXX` (nice)